src/bokeh/models/widgets/indicators.py

```diff
@@ -0,0 +1,138 @@
+#-----------------------------------------------------------------------------
+# Copyright (c) 2012 - 2023, Anaconda, Inc., and Bokeh Contributors.
+# All rights reserved.
+#
+# The full license is in the file LICENSE.txt, distributed with this software.
+#-----------------------------------------------------------------------------
+''' Various kinds of indicator widgets.
+
+'''
+
+#-----------------------------------------------------------------------------
+# Boilerplate
+#-----------------------------------------------------------------------------
+from __future__ import annotations
+
+import logging # isort:skip
+log = logging.getLogger(__name__)
+
+#-----------------------------------------------------------------------------
+# Imports
+#-----------------------------------------------------------------------------
+
+# Standard library imports
+from typing import Any
+
+# Bokeh imports
+from ...core.enums import Orientation
+from ...core.has_props import abstract
+from ...core.properties import (
+    Bool,
+    Enum,
+    Int,
+    Nullable,
+    String,
+)
+from .widget import Widget
+
+#-----------------------------------------------------------------------------
+# Globals and constants
+#-----------------------------------------------------------------------------
+
+__all__ = (
+    "Progress",
+)
+
+#-----------------------------------------------------------------------------
+# Dev API
+#-----------------------------------------------------------------------------
+
+@abstract
+class IndicatorWidget(Widget):
+    """ Abstract base class for indicator widgets.
+
+    """
+
+    # explicit __init__ to support Init signatures
+    def __init__(self, *args: Any, **kwargs: Any) -> None:
+        super().__init__(*args, **kwargs)
+
+#-----------------------------------------------------------------------------
+# General API
+#-----------------------------------------------------------------------------
+
+class Progress(IndicatorWidget):
+    """ Progress indicator widget.
+
+    .. note::
+        Due to the cooperative multi-tasking nature of JS runtime environments
+        and bokehjs being a single-threaded library, long running tasks may
+        prevent the progress from updating until computationally heavy tasks
+        are completed.
```

Comment:
I think just "Long running tasks may prevent the progress from updating until they are completed" is  all the information that users actually need to see.

Reply:
Perhaps this could be simplified, but I would expect an explanation why (is it a limitation of bokeh, web browsers or maybe both?).